File: test_parser.py
import unittest
import parser


class TestParserMethods(unittest.TestCase):

    def test_parse_log_file(self):
        with self.assertRaises(AssertionError):
            parser.parse_log_file('nofile')

        records = parser.parse_log_file('test.log')

        self.assertEqual(len(records), 30)

    def test_mean_response(self):
        records = parser.parse_log_file('test.log')
        self.assertEqual(parser.mean_response(records), 76.8)

    def test_mode_response(self):
        records = parser.parse_log_file('test.log')
        self.assertEqual(parser.mode_response(records), 18)

    def test_median_response(self):
        records = parser.parse_log_file('test.log')
        self.assertEqual(parser.median_response(records), 36.5)

    def test_mode_dyno(self):
        records = parser.parse_log_file('test.log')
        self.assertEqual(parser.mode_dyno(records), 'web.1')


if __name__ == '__main__':
    unittest.main()

File: test_parser.rb
#!/usr/bin/ruby

require 'test/unit'
require './parser'


class TestParser < Test::Unit::TestCase

  def setup
    @records = parse_log_file('test.log')
    assert_equal(@records.length, 30)
  end

  def test_mean_response
    assert_equal(mean_response(@records), 76.8)
  end

  def test_mode_response
    assert_equal(mode(@records, 'response_time'), 18)
  end

  def test_median_response
    assert_equal(median_response(@records), 36.5)
  end

  def test_mode_dyno
    assert_equal(mode(@records, 'dyno'), 'web.1')
  end

end
